### ISSUE TITLE
Edge detection

### ISSUE THREAD

**Issue by AhmedKhaled945** [lood339/SCCvSD#12]
Hello Dr.Chen, i managed to get the pipeline working, but unfortunately the edge detection models is giving disconnected lines output, with not high accuracy, so what do you propose to tune it or enhance the training a little bit?

Sorry i didn't find a way to post this on the two-gan-PyTorch repo by you, as it is forked, so i posted it here.

**Comment by lood339** [comment]
Can I have more information of "disconnected lines output"? I assume you use the soccer dataset. One way is to check if the training is converged. For example, how about the line detection on the training set?

**Comment by AhmedKhaled945** [comment]
Hello Dr.Chen, thanks for the prompt answer,
Yes i used the dataset provided by you, soccer dataset,
here is an example for my output on a training set input.
I used your command of training provided in the repo, using the latest weights.

![image](https://user-images.githubusercontent.com/36905018/88180389-97c3af00-cc2d-11ea-9550-facb7815c116.png)

![image](https://user-images.githubusercontent.com/36905018/88180418-a4480780-cc2d-11ea-9cfe-df26defa6377.png)

**Comment by AhmedKhaled945** [comment]
And on my soccer matches, it give me the following results (there are more bad examples, compeltely disconnected)

![image](https://user-images.githubusercontent.com/36905018/88180846-423bd200-cc2e-11ea-916a-a3f592abc1c9.png)

![image](https://user-images.githubusercontent.com/36905018/88180896-541d7500-cc2e-11ea-913d-56e6add7046a.png)

**Comment by lood339** [comment]
1. The edge detection in the training set does not look good. I suspect the training process does not converge. For the training set, the network should give very good result.
2. For the testing example, the edge is not connected. However, it may not heavily impact the final result. The edge image is used to query a nearest neighbour in a camera pose database. So, the edge image can be inaccurate (as the one you shown).

 My suggestion is that you can use this edge image to estimate and optimize the camera pose. Then, project the soccer court to the image using the estimated camera pose.

**Comment by AhmedKhaled945** [comment]
My problem is there are far more bad examples that the ones i've shown, this is a random one, like completely disconnected, even me as a human find it difficult to know where is this edge map located with respect to the real world pitch,
So your suggestion is we need more training, or model tuning for training to converge? or what do you think?

**Comment by AhmedKhaled945** [comment]
Can i ask how many epochs you trained the model for? the default is 200, but i don't think it is enough? is it more?
